cardano-api/src/Cardano/Api/ProtocolParameters.hs

```diff
@@ -213,14 +204,347 @@ data ProtocolParameters =
+                     Nothing Nothing
+        ShelleyBasedEraAllegra -> do
+          return $ createProtocolParameters
+                     ver decent exEntropy bHeadSize bBodySize maxTxSize
+                     txFeeFixed txFeePerByte minUtxo stakeAddrDep stakePoolDep
+                     minPoolCost poolRetireEpochMax poolTargetNum poolPledge
+                     monExpansion treasuryCut Nothing mempty Nothing Nothing
+                     Nothing Nothing
+        ShelleyBasedEraMary -> do
+          return $ createProtocolParameters
+                     ver decent exEntropy bHeadSize bBodySize maxTxSize
+                     txFeeFixed txFeePerByte minUtxo stakeAddrDep stakePoolDep
+                     minPoolCost poolRetireEpochMax poolTargetNum poolPledge
+                     monExpansion treasuryCut Nothing mempty Nothing Nothing
+                     Nothing Nothing
+     -- ShelleyBasedEraAlonzo ->
+     --   return $ createProtocolParameters
+     --              ver decent exEntropy bHeadSize bBodySize maxTxSize
+     --              txFeeFixed txFeePerByte Nothing stakeAddrDep stakePoolDep
+     --              minPoolCost poolRetireEpochMax poolTargetNum poolPledge
+     --              monExpansion treasuryCut utxoCostPerByte costModels
+     --              prices maxTxExUnits maxBlockExUnits maxValueSize
+ where
+  -- | Protocol parameters change in Alonzo. We check
+  --that the required protocol parameters are specified
+  --in the relevant era.
+  checkNecessaryParamsSpecified
+    :: ShelleyBasedEra era -> Either ProtocolParametersError ()
+  checkNecessaryParamsSpecified sbe' = do
+    let anyEra = anyCardanoEra era
+    case sbe' of
+      ShelleyBasedEraShelley ->
+         parameterExists minUtxo (ProtocolParamsErrMustDefineMinUTxo anyEra)
+      ShelleyBasedEraAllegra ->
+         parameterExists minUtxo (ProtocolParamsErrMustDefineMinUTxo anyEra)
+      ShelleyBasedEraMary ->
+         parameterExists minUtxo (ProtocolParamsErrMustDefineMinUTxo anyEra)
+  --  ShelleyBasedEraAlonzo -> do
+  --     parameterExists utxoCostPerByte (ProtocolParamsErrMustDefineCostPerByte anyEra)
+  --     parameterExists costModels (ProtocolParamsErrMustDefineCostModel anyEra)
+  --     parameterExists prices (ProtocolParamsErrMustDefineScriptExecPrices anyEra)
+  --     parameterExists maxTxExUnits (ProtocolParamsErrMustDefineMaxTxExec anyEra)
+  --     parameterExists maxBlockExUnits (ProtocolParamsErrMustDefineMaxBlockExec anyEra)
+  --     parameterExists maxValueSize (ProtocolParamsErrMustDefineMaxValueSize anyEra)
+
+  -- | Check if a protocol parameter was specified and fail if not.
+  parameterExists
+    :: Maybe a
+    -> ProtocolParametersError
+    -> Either ProtocolParametersError ()
+  parameterExists value err =
+    maybe (Left err) Right value >> Right ()
+
+
+newtype MaxTxExecutionUnits =
+    MaxTxExecutionUnits { unMaxTxExecutionUnits :: ExecutionUnits}
```

Comment:
Do we really need this and `MaxBlockExecutionUnits` as newtype wrappers over `ExecutionUnits`? It means duplicaing the ToJSON and FromJSON and other conversion functions. We have followed the pattern for the other protocol params of just using their direct types and using the record field name to distinguish one param from another, rather than giving every single param its own newtype. I think that's a sensible pattern to stick to, which would mean changing both `protocolParamMaxTxExUnits` and `protocolParamMaxBlockExUnits` fields to be simply `Maybe ExecutionUnits`.

This would involve moving the To/FromJSON instances to `ExecutionUnits` itself of course.

Reply:
It's not necessary, I'll remove it and use the ledger type directly.